### PR TITLE
Allow actor methods to return multiple object IDs.

### PR DESCRIPTION
**Pull request by robertnishihara**
Addresses #750.

This PR allows you to do

```python
@ray.remote
class Foo(object):
    def method1(self):
        return 1

    @ray.method(num_return_vals=2)
    def method2(self):
        return 1, 2
```

**TODO before merging:**
- [x] add test

### CODE REVIEW

File: python/ray/actor.py [hunk 1]
```diff
@@ -287,24 +291,29 @@ def temporary_actor_method(*xs):
...
-    for actor_method_name in actor_method_names:
+
+    for actor_method_name, num_return_vals in zip(
+            actor_method_names, actor_method_num_return_vals):
         # TODO(rkn): When we create a second actor, we are probably overwriting
         # the values from the first actor here. This may or may not be a
         # problem.
         function_id = compute_actor_method_function_id(class_name,
                                                        actor_method_name).id()
         # For now, all actor methods have 1 return value.
```
Comment (pcmoritz) [inline python/ray/actor.py]: this comment is outdated

File: python/ray/actor.py [hunk 2]
```diff
@@ -287,24 +291,28 @@ def temporary_actor_method(*xs):
...
+            actor_method_names, actor_method_num_return_vals):
         # TODO(rkn): When we create a second actor, we are probably overwriting
         # the values from the first actor here. This may or may not be a
         # problem.
         function_id = compute_actor_method_function_id(class_name,
                                                        actor_method_name).id()
-        # For now, all actor methods have 1 return value.
         worker.function_properties[driver_id][function_id] = (
-            FunctionProperties(num_return_vals=2,
+            FunctionProperties(num_return_vals=num_return_vals + 1,
```
Comment (stephanie-wang) [inline python/ray/actor.py]: Can you add a comment that the last argument is the dummy return value? I probably should've done this before :)

File: python/ray/actor.py [hunk 3]
```diff
@@ -287,24 +291,28 @@ def temporary_actor_method(*xs):
...
         worker: The worker to register the signatures on.
         driver_id: The ID of the driver that this actor is associated with.
         actor_id: The ID of the actor.
         actor_method_names: The names of the methods to register.
+        actor_method_num_return_vals: A list of the number of return values for
+            each of the actor's methods.
     """
-    for actor_method_name in actor_method_names:
+
+    for actor_method_name, num_return_vals in zip(
```
Comment (stephanie-wang) [inline python/ray/actor.py]: Maybe add an assertion that `len(actor_method_names) == len(actor_method_num_return_vals)`

**Review by stephanie-wang — COMMENTED** [review]
Hmm, is there a way to determine the number of return values automatically? Similar to what we do with normal remote functions?

EDIT: I see, so this just determines the number of object IDs returned?

**Comment by robertnishihara** [comment]
Right, it says how many object IDs to return. I don't think there is a good way of doing this automatically, and in general Python functions can return variable numbers of arguments.

File: python/ray/actor.py (hunk 3 above)
Comment (robertnishihara) [reply python/ray/actor.py]: Fixed

File: python/ray/actor.py (hunk 2 above)
Comment (robertnishihara) [reply python/ray/actor.py]: Fixed

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**